### ISSUE TITLE
Wavelength range calculated wrong by `Spectrum.get(wunit='nm')` in some cases

### ISSUE THREAD

**Issue by bfrangi** [radis/radis#707]
### 🐛 Describe the bug

The wavelength array is wrongly calculated in some cases. For instance, modifying the basic example in the README slightly to this, where `2937.6340295525983 cm-1` is equal to `3404.1 nm` and `2937.9792578664396 cm-1` is equal to `3403.7 nm`:
```python
from radis import calc_spectrum
s = calc_spectrum(2937.6340295525983, 2937.9792578664396,         # cm-1
                  molecule='CH4',
                  isotope='1',
                  pressure=1.01325,   # bar
                  Tgas=400,           # K
                  mole_fraction=0.02,
                  path_length=2,      # cm
                  databank='hitran',   # or use 'hitemp'
                  verbose=False
                  )
s.plot('transmittance_noslit', wunit='nm') 
```
Gives the following plot:

![bugradis](https://github.com/user-attachments/assets/c3a3c273-62c5-4655-8c7e-fd12e9fdb6b8)

Note that not even the labels in `cm-1` at the top of the figure are the ones requested in the code. On the other hand, if `nm` units are not requested, the spectrum is calculated correctly:

![bugradis2](https://github.com/user-attachments/assets/3391d541-2f7c-45c0-b639-58237df2cd0e)

### 💡 Possible solutions

I just discovered `radis`, so I'm not very familiar with the codebase. I have no idea as to why this happens, but at the moment I am doing the conversion between `cm-1` and `nm` myself with this function and it works very well:

```python
def wavelength_to_wavenumber(wl: float) -> float:
    """
    Convert wavelength to wavenumber.

    Parameters
    ----------
    wl : float
        Wavelength in nm.

    Returns
    -------
    float
        Wavenumber in cm^-1.
    """
    return 1 / wl * 1e7


def wavenumber_to_wavelength(wn: float) -> float:
    """
    Convert wavenumber to wavelength.

    Parameters
    ----------
    wn : float
        Wavenumber in cm^-1.

    Returns
    -------
    float
        Wavelength in nm.
    """
    return 1 / wn * 1e7
```

It could probably also be done using `astropy` quantities, but for this fits my needs nicely at the moment.

### 🎲 Radis version

0.15.2

### 💻 Operating system

Linux

**Comment by tactipus** [comment]
This might be related to this other issue #581

**Comment by erwanp** [comment]
@bfrangi can it be because of air/Vacuum ?

**Comment by bfrangi** [comment]
@erwanp Not sure if that can be the underlying reason for the bug in the *implementation*, but from a *physical* point of view that should not cause any differences, as wavenumber-wavelength conversion does not depend on any property of the medium... Note that the two graphs that I posted in the issue are for exactly the same conditions, the only difference being that `wunit='nm'` has been specified in the first case. Both plots have wavelength and wavenumber scales but in the first plot the scales are shifted with respect to the second plot.

**Comment by erwanp** [comment]
Hello @bfrangi wavenumber wavelength conversion depend on the medium.  Wavenumber is medium-independant, that's energy. Wavelength depends on the photon velocity which is not the same in air vacuum or water. 

In vacuum the conversion is obviously 1e7/wavenumber, but in air or any other medium you have to account for the dispersion index, as done by Radis using the Cidder formula https://radis.readthedocs.io/en/latest/source/radis.phys.air.html#radis.phys.air.vacuum2air
This induced shifts of typically 0.2 nm in the visible range, which might or might not be relevant depending on your application.

In Radis you can specify wunit =nm_air or wunit=nm_vac to be explicit, we had to arbitrary decide what would the usual 'nm'  be and decided that is would correspond to 'nm_air' because Radis was mainly used for atmospheric applications, but of course it might be different in your use case

**Comment by bfrangi** [comment]
@erwanp you are right, wavenumber-wavelength conversion *does* depend on the medium because wavelength depends on the medium. My mistake, thanks for clarifying.

As for specifying the units, I have seen that in `calc_spectrum(wunit=)` the only units you can specify are `'nm'` and `'cm-1'`, but not `'nm_air'` or `'nm_vac'`. These last two raise `ValueError: 'nm_air' did not parse as unit`. The `plot(wunit=)` function *does* accept these medium-specific units, however.

What I would suggest is to add the medium-specific units to the `calc_spectrum(wunit=)` function also. That way, this:
```python
from radis import calc_spectrum
s = calc_spectrum(3403.7, 3404.1,         # cm-1
                  molecule='CH4',
                  isotope='1',
                  pressure=1.01325,   # bar
                  Tgas=400,           # K
                  mole_fraction=0.02,
                  path_length=2,      # cm
                  databank='hitran',   # or use 'hitemp'
                  verbose=False,
                  wunit='nm_vac' 
                  )
s.plot(wunit='nm_vac')
```
would give the expected results.

The way it is done now, if I want to compute the spectrum between two wavelengths in vacuum, I have to convert the wavelengths from vacuum to wavelengths in air, and then use those in the `calc_spectrum()` function, only to then have to specify `plot(wunit='nm_vac')` for the plot, which involves too many conversions and units in the process which make it very confusing. Adding the medium-specific unit options to `calc_spectrum()` would simplify things a lot.

**Comment by erwanp** [comment]
> **EDIT**: I just saw the `medium` parameter in the `calc_spectrum()` function, and I see that it can be used to specify what medium the wavelength range in that function is given in. However, when using the `plot()` function, having specified the medium as `'vacuum'` in the `calc_spectrum()` function, the results are still given as if the spectrum had been computed in air. Wouldn't it make more sense for `plot()` to take the units based on the medium specified in the `calc_spectrum()` function, instead of assuming always that it is `'nm_air'`?
> 
> It would be like this:
> 
> * If the user specified `medium='air'` (or did not specify) in `calc_spectrum()`, then all plots are done taking `'nm_air'` as default.
> * If the user specified `medium='vacuum'` in `calc_spectrum()`, then all plots are done taking `'nm_vac'` as default.
> * There is no need to accept `'nm_vac'` and `'nm_air'` in the `calc_spectrum()` function, as the medium can already be specified with the `medium` parameter.


I agree with this suggestion, output range should be consistent with the medium parameter  used, and the 'nm_vac'  and 'nm_air'  should also be usable as 'wunit'  arguments to be more consistent. Modifications should take place in the `get_wavenumber_range` function called by calc_spectrum here https://github.com/radis/radis/blob/072d8209e3e6cc39afda55b0e93fbaf2119eb1ac/radis/lbl/calc.py#L421
Do you want to try to implement this?

**Comment by bfrangi** [comment]
Sure @erwanp, I can have a look at this. To summarize the discussion and document the issue now that I understand it better, here are a few code examples:

**Calculating the spectrum in vacuum and not specifying units at plot time**
```python
from radis import calc_spectrum
s = calc_spectrum(3403, 3404,
                  wunit='nm',
                  medium='vacuum',
                  molecule='CH4',
                  Tgas=400,
                  mole_fraction=0.1)
s.plot()
```
This should give the lower x axis from `3403 nm` to `3404 nm` and the top axis from `2938.58 cm-1` to `2937.72 cm-1`. Instead, this gives the correct lower x axis range from `3403 nm` to `3404 nm` but the top axis wavenumber is computed as if the spectrum was calculated in air, from `2937.8 cm-1` to `2936.92 cm-1`: [see plot](https://github.com/user-attachments/files/17965415/vacuum-calc-default-plot.pdf).

**Calculating the spectrum in vacuum and specifying `nm` units at plot time**
```python
from radis import calc_spectrum
s = calc_spectrum(3403, 3404,
                  wunit='nm',
                  medium='vacuum',
                  molecule='CH4',
                  Tgas=400,
                  mole_fraction=0.1)
s.plot(wunit='nm')
```
This should give the lower x axis from `3403 nm` to `3404 nm` and the top axis from `2938.58 cm-1` to `2937.72 cm-1`. Instead, this gives the lower x axis range from `3402.06 nm` to `3403.07 nm` and the top axis wavenumber is computed as if the spectrum was calculated in vacuum but using the incorrect values of wavelength, from `2939.4 cm-1` to `2938.52 cm-1`: [see plot](https://github.com/user-attachments/files/17965433/vacuum-calc-nm-plot.pdf).

**Calculating the spectrum in vacuum and specifying `cm-1` units at plot time**
```python
from radis import calc_spectrum
s = calc_spectrum(3403, 3404,
                  wunit='nm',
                  medium='vacuum',
                  molecule='CH4',
                  Tgas=400,
                  mole_fraction=0.1)
s.plot(wunit='cm-1')
```
This should give the lower x axis from `2938.58 cm-1` to `2937.72 cm-1` and the top axis from `3403 nm` to `3404 nm`. This plot is calculated correctly: [see plot](https://github.com/user-attachments/files/17965451/vacuum-calc-cm-1-plot.pdf).

**Comment by erwanp** [comment]
We're making progress, thanks for these examples. Took the libery of numbering them. 

####  Case 1.  I understand that the secondary-axis (upper axis) is faulty. 

This is a newer part of the code; and might have been implemented wrong. 
2nd axis was asked for here https://github.com/radis/radis/issues/446 and implemented there https://github.com/radis/radis/pull/537 (Jan 2023) 

In [this line L2168](https://github.com/radis/radis/pull/537/files#diff-4fd1b50748cf33b23ca976c2da5650965761df602d15219ffa4c6d5dd2e1699eR2168) we add a secondary (upper) wavelength axis assuming medium is vacuum.
- We shall instead use the Spectrum medium ; **Problem** medium is not stored anymore in the Spectrum object since 2019 (https://github.com/radis/radis/pull/51), that information is in the wavespace used ("nm_air", "nm_vac", "cm-1"); but in Example 1 it's "cm-1" therefore it loses the information  ;   
- [ ] We shall therefore just use the same Convention in the secondary axis as in the primary axis and consider that "nm" is "nm_air" and not "nm_vac"; the fix is by changing the conversion functions in [Line 2169](https://github.com/radis/radis/pull/537/files#diff-4fd1b50748cf33b23ca976c2da5650965761df602d15219ffa4c6d5dd2e1699eR2169) from "cm2nm", "nm2cm" to "cm2nm_air", "nm_air2cm"). Maybe also setting the secondary axis label to `secx.set_xlabel('Wavelength (nm) [air]')`   to avoid any furthe confusion

Note : plot() has a "[show_medium](https://github.com/radis/radis/blob/072d8209e3e6cc39afda55b0e93fbaf2119eb1ac/radis/spectrum/spectrum.py#L2124)" parameter which is used to show you "nm [vac]" and "nm [air]" in the label; diffusing any confusion. 
- [ ] Maybe we shall make it True by default ?  



#### Case 2. 

Just to be clear can you re-run it with s.plot(wunit="nm_vac") . It might give you the expected results, and the confusion might come from our convention that "nm" refers to "nm_air" instead of being "nm in the medium in which the spectrum was requested". 

In that case I'd also apply the Note above, showing the medium as default, to avoid any further confusion
